sql/core/src/test/scala/org/apache/spark/sql/execution/PlannerSuite.scala

```diff
@@ -994,6 +994,20 @@ class PlannerSuite extends SharedSparkSession with AdaptiveSparkPlanHelper {
       }
     }
   }
+
+  test("Change the number of partitions to zero when a range is empty") {
+    withSQLConf(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "true") {
```

Comment:
Could you use `testWithWholeStageCodegenOnAndOff` instead?

Reply:
Oh, I didn't notice I can use it. Thanks.